unitTests/test_helper_functions.cpp

```diff
@@ -19,3 +24,47 @@ TEST(string_from_unterminated, unterminatedArray)
     ASSERT_EQ(res.size(), 4u);
     ASSERT_STREQ(res.c_str(), "abcd");
 }
+
+struct WideStr2Utf8Test : public ::testing::Test
+{
+    WideStr2Utf8Test()
+    {
+        if (std::setlocale(LC_ALL, "en_US.UTF-8") == nullptr) {
```

Comment:
I think it would be better here to do:

```cpp
ASSERT_NE(nullptr, std::setlocale(LC_ALL, "en_US.UTF-8"));
```
In case the test fails, we would get a very clear message. You could also use `<<` to provide a string in case of error if the message would not be clear enough.

Reply:
That doesn't work, `ASSERT_NE()` want's to return a value but this is a constructor.

Reply:
ahg right, I forgot that thing about Gtest.